src/workerd/api/headers.c++

```diff
@@ -98,6 +98,87 @@ void requireValidHeaderValue(kj::StringPtr value) {
+// to send.
+static constexpr size_t MAX_COMMON_HEADER_ID =
+    static_cast<size_t>(capnp::CommonHeaderName::WWW_AUTHENTICATE);
+
+// Constexpr array of lowercase common header names (must match CommonHeaderName enum order
+// and must be kept in sync with the ordinal values defined in http-over-capnp.capnp). Since
+// it is extremely unlikely that those will change often, we hardcode them here for runtime
+// efficiency.
+static constexpr const char* COMMON_HEADER_NAMES[] = {
+  nullptr,                        // 0: invalid
+  "accept-charset",               // 1
+  "accept-encoding",              // 2
+  "accept-language",              // 3
+  "accept-ranges",                // 4
+  "accept",                       // 5
+  "access-control-allow-origin",  // 6
+  "age",                          // 7
+  "allow",                        // 8
+  "authorization",                // 9
+  "cache-control",                // 10
+  "content-disposition",          // 11
+  "content-encoding",             // 12
+  "content-language",             // 13
+  "content-length",               // 14
+  "content-location",             // 15
+  "content-range",                // 16
+  "content-type",                 // 17
+  "cookie",                       // 18
+  "date",                         // 19
+  "etag",                         // 20
+  "expect",                       // 21
+  "expires",                      // 22
+  "from",                         // 23
+  "host",                         // 24
+  "if-match",                     // 25
+  "if-modified-since",            // 26
+  "if-none-match",                // 27
+  "if-range",                     // 28
+  "if-unmodified-since",          // 29
+  "last-modified",                // 30
+  "link",                         // 31
+  "location",                     // 32
+  "max-forwards",                 // 33
+  "proxy-authenticate",           // 34
+  "proxy-authorization",          // 35
+  "range",                        // 36
+  "referer",                      // 37
+  "refresh",                      // 38
+  "retry-after",                  // 39
+  "server",                       // 40
+  "set-cookie",                   // 41
+  "strict-transport-security",    // 42
+  "transfer-encoding",            // 43
+  "user-agent",                   // 44
+  "vary",                         // 45
+  "via",                          // 46
+  "www-authenticate",             // 47
+};
+
+kj::String getCommonHeaderName(uint id) {
+  KJ_ASSERT(id > 0 && id <= MAX_COMMON_HEADER_ID, "Invalid common header ID");
+  auto name = COMMON_HEADER_NAMES[id];
+  KJ_DASSERT(name != nullptr);
+  return kj::str(name);
+}
+
+kj::Maybe<uint> getCommonHeaderId(kj::StringPtr name) {
+  // It really shouldn't be possible for a name to be empty but just in case...
+  if (name.size() == 0) return kj::none;
+  for (uint i = 1; i <= MAX_COMMON_HEADER_ID; ++i) {
```

Comment:
Previously we used a HashMap here, but you've replaced it with a linear search? I realize the list is not all that big but I'd still expect the HashMap to be much faster. Why did you remove it?